andela/balder-ah-backend
# #163016225 Retrieve all articles with author's profile

#### What does this PR do?
This PR enables users retrieve their articles with their profile and also retrieve the articles of other authors with their profiles


#### Description of Task to be completed?
 - Update current user controller to also retrieve user's articles
 - Update controller to retrieve other author's profile to also retrieve their articles


#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a POST request to **localhost:3000/api/articles** to post a new article with the required fields (title, description, body, tags(should be an array))
- Send a GET request to **localhost:3000/api/user** to get your profile details
- Send a GET request to **localhost:3000/api/profiles/:username** to get another author's profile details
- Send a PUT request to **localhost:3000/api/user** with the required fields (username, email, bio, image) filled to update your profile details
- Send a GET request to **localhost:3000/api/profiles/(enter username)** to retrieve a user's profile
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**



#### Any background context you want to provide?
N/A


#### What are the relevant pivotal tracker stories? (If applicable)
[#163016225](https://www.pivotaltracker.com/story/show/163016225)


#### Screenshots (If applicable)

<img width="1440" alt="screen shot 2019-01-08 at 9 00 37 am" src="https://user-images.githubusercontent.com/29441898/50817412-04b17580-1324-11e9-968c-87b5a2d74f1e.png">
<img width="1440" alt="screen shot 2019-01-08 at 9 01 01 am" src="https://user-images.githubusercontent.com/29441898/50817414-04b17580-1324-11e9-985f-148eb7bc6bca.png">


#### Questions:
